### PR TITLE
[AQUA][Evaluate] Added the EvaluationServiceConfig class to manage evaluation service configurations.

### PR DESCRIPTION
**Pull request by mrDzurb**
## Description
Added the `EvaluationServiceConfig` class along with supporting structures to manage evaluation service configurations. This includes methods to merge evaluation inference parameters, retrieve model parameters based on framework and version, and filter shapes based on provided criteria. The implementation ensures flexibility and scalability for various frameworks and tasks.

## Dependency
This PR has a dependency on the [Introduce Global Configuration for LLM Evaluation](https://bitbucket.oci.oraclecorp.com/projects/ODSC/repos/odsc_environments/pull-requests/969/overview)

## Tests
image

### CODE REVIEW

File: ads/aqua/config/evaluation/evaluation_service_config.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+                evaluation_container
+                and evaluation_container not in shape.filter.evaluation_container
+            ):
+                continue
+            if (
+                evaluation_target
+                and evaluation_target not in shape.filter.evaluation_target
+            ):
+                continue
+            results.append(shape)
```
Comment (mayoor) [inline ads/aqua/config/evaluation/evaluation_service_config.py:239]: Can this be - 
```
key = evaluation_container or evaluation_target
if key in shape.filter.evaluation_container:
    results.append(shape)
```

File: ads/aqua/config/evaluation/evaluation_service_config.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+    versions: Optional[Dict[str, ModelParamsVersion]] = Field(default_factory=dict)
+
+    class Config:
+        extra = "ignore"
+
+
+class InferenceParams(Serializable):
+    """Contains inference-related parameters with defaults."""
+
+    inference_rps: Optional[int] = INFERENCE_RPS
```
Comment (mayoor) [inline ads/aqua/config/evaluation/evaluation_service_config.py:57]: Can we move these defaults to the config json? Maybe they are already part of the config json and these initial values get overwritten. Not clear from the code, hence checking.

Comment (mrDzurb) [reply ads/aqua/config/evaluation/evaluation_service_config.py:57]: Yeah, let me do this. These defaults are placed in the config as well.

File: ads/aqua/config/evaluation/evaluation_service_config.py (hunk 1 above)
Comment (mrDzurb) [reply ads/aqua/config/evaluation/evaluation_service_config.py]: Done. Having one key would not work, however i simplified a logic a bit.

File: ads/aqua/config/evaluation/evaluation_service_config.py (hunk 2 above)
Comment (mrDzurb) [reply ads/aqua/config/evaluation/evaluation_service_config.py]: Done.

**Review by mayoor — APPROVED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -74,6 +74,7 @@ dependencies = [
   "scikit-learn>=1.0",
   "tabulate>=0.8.9",
   "tqdm>=4.59.0",
+  "pydantic>=2.6.3",
```
[the remark this answers is not in the document]
Comment (mrDzurb) [reply pyproject.toml:77]: Good point, will add.

File: ads/aqua/config/utils/serializer.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+            and "signer" not in kwargs
+        ):
+            kwargs.update(default_signer())
+        with fsspec.open(uri, "r", **kwargs) as f:
+            return f.read()
+
+    def to_json(
+        self,
+        uri: str = None,
+        encoder: callable = json.JSONEncoder,
```
[the remark this answers is not in the document]
Comment (mrDzurb) [reply ads/aqua/config/utils/serializer.py:95]: Wasn't aware of this, will take a look. Will play with this.

**Pull request merged**